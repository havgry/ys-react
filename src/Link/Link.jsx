import React from 'react'
import PropTypes from 'prop-types'
import classNames from 'classnames'

import Anchor from '../helpers/Anchor'

import '../base.css'
import '@youseedk/dna/css/elements/ys-link.css'

const Link = ({
  href,
  to,
  children,
  className,
  ...props
}) => (
  <Anchor
    href={href}
    to={to}
    className={classNames(
      'ys-link',
      className,
    )}
    {...props}
  >
    {children}
  </Anchor>
)

Link.propTypes = {
  children: PropTypes.node.isRequired,
  to: PropTypes.string,
  href: PropTypes.string,
  className: PropTypes.string,
}

Link.defaultProps = {
  to: null,
  href: null,
  className: '',
}

export default Link
