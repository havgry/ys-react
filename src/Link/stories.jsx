import React from 'react'
import { storiesOf } from '@storybook/react'
import StoryRouter from 'storybook-react-router'

import { text } from '@storybook/addon-knobs'

import Link from './Link'

storiesOf('Link', module)
  .addParameters({ jest: ['Link'] })
  .addDecorator(StoryRouter())
  .add('Anchor element', () => (
    <Link
      href={text('href', 'https://yousee.dk', 'Content')}
      className={text('className', '', 'Misc')}
    >
      External link
    </Link>
  ))
  .add('React Router <NavLink>', () => (
    <Link
      to={text('to', '/dna', 'Content')}
      className={text('className', '', 'Misc')}
    >
      Internal link
    </Link>
  ))
