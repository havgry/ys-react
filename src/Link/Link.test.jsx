import React from 'react'
import { shallow, mount } from 'enzyme'

import Link from './Link'

describe('<Link>', () => {
  describe('Props', () => {
    it('Throws propType warning if no `children` prop is passed', () => {
      expect(
        () => shallow(
          <Link href="https://yousee.dk" />,
        ),
      ).toThrow()
    })

    it('Has correct defaultProps', () => {
      const {
        className,
        href,
        to,
      } = mount(
        // eslint-disable-next-line jsx-a11y/anchor-is-valid
        <Link>
          YouSee.dk
        </Link>,
      ).props()

      expect(className).toBe('')
      expect(href).toBe(null)
    })
  })

  describe('Rendering', () => {
    it('Custom `className` is being applied to the container', () => {
      expect(
        shallow(
          <Link
            href="https://yousee.dk"
            className="salamander"
          >
            YouSee.dk
          </Link>,
        )
          .find('.ys-link.salamander')
          .exists(),
      ).toBe(true)
    })

    it('Forwards other generic props to container element', () => {
      const props = shallow(
        <Link
          href="https://yousee.dk"
          target="_blank"
          title="Gå til YouSee.dk"
        >
          YouSee.dk
        </Link>,
      ).props()

      expect(props.title).toBe('Gå til YouSee.dk')
      expect(props.target).toBe('_blank')
    })

    it('Doesn\'t add `ys-link` class if given `unstyled` prop', () => {
      expect(
        shallow(
          <Link
            unstyled
            href="https://yousee.dk"
          >
            YouSee.dk
          </Link>,
        )
          .find('.ys-link')
          .exists(),
      ).toBe(false)
    })
  })

  describe('Snapshots', () => {
    it('With default props', () => {
      expect(
        shallow(
          <Link to="/kurv">
            Til kurv
          </Link>,
        ),
      ).toMatchSnapshot()
    })

    it('With custom props', () => {
      expect(
        shallow(
          <Link
            href="https://yousee.dk"
            target="_blank"
            title="Gå til YouSee.dk"
          >
            YouSee.dk
          </Link>,
        ),
      ).toMatchSnapshot()
    })
  })
})
