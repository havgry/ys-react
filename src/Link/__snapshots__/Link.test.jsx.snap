// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link> Snapshots With custom props 1`] = `
<Anchor
  className="ys-link"
  disabled={false}
  href="https://yousee.dk"
  target="_blank"
  title="GÃ¥ til YouSee.dk"
  to={null}
>
  YouSee.dk
</Anchor>
`;

exports[`<Link> Snapshots With default props 1`] = `
<Anchor
  className="ys-link"
  disabled={false}
  href={null}
  to="/kurv"
>
  Til kurv
</Anchor>
`;
